cannon/mipsevm/testutil/arch.go

```diff
@@ -42,6 +44,21 @@ func SetMemoryUint64(t require.TestingT, mem *memory.Memory, addr Word, value ui
 	require.Equal(t, Word(value), actual)
 }
 
+// SetMemoryUint32 writes a 4-byte value to memory.
+// Also, randomizes the rest of the Word containing the uint32
+func SetMemoryUint32(mem *memory.Memory, addr Word, val uint32, randomizeWordSeed int64) {
```

Comment:
nit: worth using a more descriptive function name that notes it randomizes the word. It's easy to get it mixed up with `SetMemoryUint64` which doesn't randomize the word written.